### PR TITLE
[http] Fix H/1 test issue in codec fuzzer

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Commit Message: Fix codec fuzz bug when encoding response headers that contain upgrade headers. The HCM scrubs these [here](https://github.com/envoyproxy/envoy/blob/6246920219ac0ba215bfd0b5462ef78036363b75/source/common/http/conn_manager_impl.cc#L1396) , but without this, the fuzz encoder mistakenly believes the response is not chunk encoded because of the presence of upgrade headers.
Risk Level: Low
Testing: Added fuzzing regression test and also added an H/1 integration test to show that behavior is good as is.
Fixes:
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=22901

### CODE REVIEW

File: test/integration/integration_test.cc [hunk 1]
```diff
@@ -1550,6 +1550,36 @@ TEST_P(IntegrationTest, TestUpgradeHeaderInResponse) {
   EXPECT_EQ("Hello World", response->body());
 }
 
+TEST_P(IntegrationTest, TestUpgradeHeaderInResponseWithTrailer) {
```
Comment (htuch) [inline test/integration/integration_test.cc:1553]: Can you add a short `//` above this explaining what it demonstrates for posterity?

Comment (htuch) [inline test/integration/integration_test.cc:1553]: IIUC the idea is to validate that upgrade is removed if the request is missing upgrade, but the rest of the trailers and unrelated headers are preserved?

**Review by htuch — COMMENTED** [review]
Thanks for the fix, this shows the limitation of having a fuzzer that is duplicating HCM behavior, I think if I was writing it from scratch again this would probably suggest a coarser decomposition for fuzzing and have HCM+codecs fuzzed together.

File: test/integration/integration_test.cc (hunk 1 above)
Comment (asraa) [reply test/integration/integration_test.cc]: Yes that's correct! Added a comment.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**